### ISSUE TITLE
dynr installation issues 2023

### ISSUE THREAD

**Issue by symiin** [mhunter1/dynr#276]
I am starting a list of installation issues on various platforms:

1. Mac Monterey. R version = "R version 4.2.2 (2022-10-31)"
Error: object ‘shortPathName’ is not exported by 'namespace:utils'
Execution halted
ERROR: lazy loading failed for package ‘dynr’
* removing ‘/private/var/folders/rq/51l3j2n56c11zfmf16bqqqg92wm6cf/T/RtmpV8Xjj1/Rinst713b7537d2b6/dynr’
      -----------------------------------
ERROR: package installation failed

**Comment by symiin** [comment]
Windows 11, R.Version = "R version 4.2.2 (2022-10-31 ucrt)"
dynr working before but recently shows this error
C:\rtools42\x86_64-w64-mingw32.static.posix\bin/ld.exe: C:/R/local323/lib/x64/libgsl.a(error.o):error.c:(.text+0x3b): undefined reference to `__imp___iob_func'

**Comment by jeksterslab** [comment]
> Windows 11, R.Version = "R version 4.2.2 (2022-10-31 ucrt)" dynr working before but recently shows this error C:\rtools42\x86_64-w64-mingw32.static.posix\bin/ld.exe: C:/R/local323/lib/x64/libgsl.a(error.o):error.c:(.text+0x3b): undefined reference to `__imp___iob_func'

I think the issue here is the incompatibility of toolchain in the newer versions of `R`. I tried installing `dynr` in Windows using the compatible `Rtools` version. I totally abandoned `local323` and relied on the headers and binaries produced by `Rtools`.

**Comment by mhunter1** [comment]
> I replaced importFrom("utils", "str", ".DollarNames", "packageVersion", "shortPathName") with import(utils) in the NAMESPACE file and added utils under Imports in the DESCRIPTION file. If this fix is acceptable I can implement it on the master branch.

@jeksterslab No, Do not do this.  I have the beginning of an alternative fix committed here: d30054fb1800f2b65a765f1db3ec2c4b35e297e7

**Comment by jeksterslab** [comment]
> > I replaced importFrom("utils", "str", ".DollarNames", "packageVersion", "shortPathName") with import(utils) in the NAMESPACE file and added utils under Imports in the DESCRIPTION file. If this fix is acceptable I can implement it on the master branch.
> 
> @jeksterslab No, Do not do this. I have the beginning of an alternative fix committed here: [d30054f](https://github.com/mhunter1/dynr/commit/d30054fb1800f2b65a765f1db3ec2c4b35e297e7)

@mhunter1 Sure. I only implemented my fix on my sandbox builds. From here on I will base my builds on your latest commit.
